### PR TITLE
BSRD-101: extend docs

### PR DESCRIPTION
**Pull request by tschanzt**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -4,14 +4,111 @@
 .. image:: https://travis-ci.org/camptocamp/marabunta.svg?branch=master
     :target: https://travis-ci.org/camptocamp/marabunta
 
-Marabunta is a name given to the migration of the legionary ants or to the ants
-themselves. Restless, they eat and digest everything in their way.
```
Comment (guewen) [inline README.rst]: Can we keep this one, maybe in italic?

File: README.rst [hunk 2]
```diff
@@ -4,14 +4,111 @@
...
+--allow-serie                MARABUNTA_ALLOW_SERIE      Allow multiple versions to be upgraded at once.
+--force-version              MARABUNTA_FORCE_VERSION    Force the upgrade to a version no matter what.
+--web-host                   MARABUNTA_WEB_HOST         Interface to bind (defaults to 0.0.0.0).
+--web-port                   MARABUNTA_WEB_PORT         Port for the Webhost (defaults to 8069).
+--web-custom-html            MARABUNTA_WEB_CUSTOM_HTML  Path to custom html page to serve.
+=================  ========  =========================  ===============================================
+
+YAML layout & Example
+=====================
+Here is an Example migration file::
```
Comment (guewen) [inline README.rst]: Any possibility to put this in a standalone file that we could reference from here and read from parser.py?

File: README.rst [hunk 3]
```diff
@@ -4,14 +4,111 @@
...
-This tool aims to run migrations for Odoo versions as efficiently as a
-Marabunta migration.
 
-It loads migration instructions from a YAML file and run the operations if
-required.
+Usage
+=====
+After installing marabunta will be available as a console command. To run properly it requires a migration file, which defines what should be updated/executed, and of course stuff like database pw. Most of these also can also be provided by an envvar and will be discussed in the options section.
+
+It is very much recomended to configure it, so that marabunta is ran automatically if odoo is started.
```
Comment (guewen) [inline README.rst]: At each run, marabunta will verify the versions from the migration file and process new ones. It is very much recommended to ...

File: README.rst [hunk 4]
```diff
@@ -4,14 +4,111 @@
...
+For instance adding it to your docker entrypoint.
+
+Features
+========
+
+* backup: Marabunta allows for a backup command to be executed before the migration.
+* addon upgrades: Marabunta is able to install or upgrade odoo addons.
+* addon uninstall: Addons which were previously installed can be removed with marabunta.
+* operations: Allows to execute commands before or after upgrading modules.
+* modes: Modes allow the user to execute commands only on a certain environment. i.e. creation of demodata on a dev system.
```
Comment (guewen) [inline README.rst]: maintenance page: publish an html page during the migration

File: README.rst [hunk 5]
```diff
@@ -4,14 +4,111 @@
...
+--migration-file   -f        MARABUNTA_MIGRATION_FILE   Definition file for the migration.
+--database         -d        MARABUNTA_DATABASE         Database we want to run the migration on.
+--db-user          -u        MARABUNTA_DB_USER          Database user.
+--db-password      -w        MARABUNTA_DB_PASSWORD      Database password.
+--db-port          -p        MARABUNTA_DB_PORT          Database port (defaults to 5432).
+--db-host          -H        MARABUNTA_DB_HOST          Database port (defaults to localhost).
+--mode                       MARABUNTA_MODE             Mode marabunta runs in for different envs.
+--allow-serie                MARABUNTA_ALLOW_SERIE      Allow multiple versions to be upgraded at once.
+--force-version              MARABUNTA_FORCE_VERSION    Force the upgrade to a version no matter what.
+--web-host                   MARABUNTA_WEB_HOST         Interface to bind (defaults to 0.0.0.0).
```
Comment (guewen) [inline README.rst]: "for the publishing of the maintenance page"

File: README.rst [hunk 6]
```diff
@@ -4,14 +4,111 @@
...
+--web-port                   MARABUNTA_WEB_PORT         Port for the Webhost (defaults to 8069).
+--web-custom-html            MARABUNTA_WEB_CUSTOM_HTML  Path to custom html page to serve.
```
Comment (guewen) [inline README.rst]: "maintenance html page"

File: README.rst [hunk 7]
```diff
@@ -4,14 +4,111 @@
...
+* addon upgrades: Marabunta is able to install or upgrade odoo addons.
+* addon uninstall: Addons which were previously installed can be removed with marabunta.
+* operations: Allows to execute commands before or after upgrading modules.
+* modes: Modes allow the user to execute commands only on a certain environment. i.e. creation of demodata on a dev system.
+
+Versioning systems
+------------------
+Currently Marabunta allows for two different Versioning systems:
+The classic Major.Minor.Bugfix and the Five digits long versions for OdooMajor.OdooMinor.Major.Minor.Bugfix.
+Although the first marabunta version should always be called setup. This is a standin for all the versions till the release, to make it easier to keep marabunta versions and tags in sync.
```
Comment (guewen) [inline README.rst]: This one probably needs a rewording: "This is a standin for all the versions till the release, to make it easier to keep marabunta versions and tags in sync.". I'm not sure to understand it :)

**Comment by guewen** [comment]
Good start, thanks :)

File: README.rst [hunk 8]
```diff
@@ -4,14 +4,111 @@
...
+Marabunta is used to provide an easy way to provide updates to Odoo which can be created fast and run easily. It also allows to differentiate between different environment to provide for instance demodata.
 
-This tool aims to run migrations for Odoo versions as efficiently as a
-Marabunta migration.
 
-It loads migration instructions from a YAML file and run the operations if
-required.
+Usage
+=====
+After installing marabunta will be available as a console command. To run properly it requires a migration file, which defines what should be updated/executed, and of course stuff like database pw. Most of these also can also be provided by an envvar and will be discussed in the options section.
```
Comment (simahawk) [inline README.rst]: To run properly it requires a migration file (which defines what should be updated/executed) and odoo's connection parameters (view options here -> [options section link])

File: README.rst (hunk 7 above)
Comment (simahawk) [reply README.rst]: The 1st version must be [underlined] "setup", as you define here the initial setup of your instance (read more about the rationale here [link to changelog])

**Comment by tschanzt** [comment]
@guewen @simahawk updated.

**Comment by tschanzt** [comment]
@simahawk can you take another look?

File: README.rst (hunk 3 above)
Comment (blaggacao) [reply README.rst]: "also can also" -> "can also"

Comment (grindtildeath) [reply README.rst]: s/recomended/recommended

File: README.rst [hunk 9]
```diff
@@ -4,14 +4,71 @@
...
+themselves. Restless, they eat and digest everything in their way.*
 
-This tool aims to run migrations for Odoo versions as efficiently as a
-Marabunta migration.
+Marabunta is used to provide an easy way to provide updates to Odoo which can be created fast and run easily. It also allows to differentiate between different environment to provide for instance demodata.
+
+
+Usage
+=====
+After installing marabunta will be available as a console command. To run properly it requires a migration file () which defines what should be updated/executed) and odoos connection parameters (view options in the options section.
```
Comment (grindtildeath) [inline README.rst]: "After installing marabunta**, it** will be available as a console command."

Comment (grindtildeath) [inline README.rst]: "which defines what **has to** be updated/executed"

Comment (grindtildeath) [inline README.rst]: odoo **'** s connection parameters (**cf** options section **)**.

File: README.rst (hunk 3 above)
Comment (grindtildeath) [reply README.rst]: s/ran/run

File: README.rst [hunk 10]
```diff
@@ -4,14 +4,71 @@
...
+For instance adding it to your docker entrypoint.
+
+Features
+========
+
+* backup: Marabunta allows for a backup command to be executed before the migration.
+* addon upgrades: Marabunta is able to install or upgrade odoo addons.
+* addon uninstall: Addons which were previously installed can be removed with marabunta.
+* operations: Allows to execute commands before or after upgrading modules.
+* modes: Modes allow the user to execute commands only on a certain environment. i.e. creation of demodata on a dev system.
```
Comment (grindtildeath) [inline README.rst]: s/i.e./e.g.

File: README.rst (hunk 9 above)
Comment (tschanzt) [reply README.rst]: @grindtildeath cf?

**Comment by tschanzt** [comment]
@grindtildeath addressed @blaggacao this section was already changed. can you check again?

File: README.rst (hunk 9 above)
Comment (grindtildeath) [reply README.rst]: Trailing parenthesis still missing. (and option is repeated)

File: README.rst [hunk 11]
```diff
@@ -4,14 +4,71 @@
...
+At each run marabunta verifies the versions from the migration file and and processes new ones.
+It is very much recommended to configure it, so that marabunta is ran automatically if odoo is started.
+For instance adding it to your docker entrypoint.
+
+Features
+========
+
+* backup: Marabunta allows for a backup command to be executed before the migration.
+* addon upgrades: Marabunta is able to install or upgrade odoo addons.
+* addon uninstall: Addons which were previously installed can be removed with marabunta.
```
Comment (arkostyuk) [inline README.rst]: is that true that Marabunta is capable of removing addons w/o anthem?

**Comment by simahawk** [comment]
build is still red because of RST ;)

File: README.rst [hunk 12]
```diff
@@ -4,14 +4,71 @@
...
+Versioning systems
+------------------
+Currently Marabunta allows for two different Versioning systems:
+The classic Major.Minor.Bugfix and the Five digits long versions for OdooMajor.OdooMinor.Major.Minor.Bugfix.
+Although the first marabunta version must be **setup** for the initial setup of your instance. (Find out more about the rationale here <https://github.com/camptocamp/marabunta/commit/9b96acaff8e7eecbf82ff592b7bb927b4cd82f02>)
+
+
+Options
+=======
+    +-------------------+----------+---------------------------+-------------------------------------------------------------------+
```
[the remark this answers is not in the document]
Comment (tschanzt) [reply README.rst]: @arkostyuk sadly i had to revert your changes. The issue is that the table markup between markdown(github) and rst (pypi) is not the same so i used this format which sorta looks like a table already even though it works on rst only and indented it so its a code block. removing this breaks the table completely

File: README.rst (hunk 11 above)
Comment (tschanzt) [reply README.rst]: @arkostyuk you're right. It is in the example file and has functions to handle it but they raise an error that it is not possible.

**Pull request merged**